Source/WebSocket_Base.h

```diff
@@ -7,7 +7,7 @@ namespace UrlLib
     class WebSocket::ImplBase
     {
     public:
-    ImplBase(std::string url, std::function<void()> onOpen, std::function<void()> onClose, std::function<void(std::string)> onMessage, std::function<void()> onError)
+    ImplBase(std::string url, std::function<void()> onOpen, std::function<void(int, std::string)> onClose, std::function<void(std::string)> onMessage, std::function<void(std::string)> onError)
```

Comment:
I missed this from before, but input strings should be `const std::string&` to avoid copying. Same goes for callbacks.

```suggestion
    ImplBase(const std::string& url, std::function<void()> onOpen, std::function<void(int, const std::string&)> onClose, std::function<void(const std::string&)> onMessage, std::function<void(const std::string&)> onError)
```